File: clients/java_graal/src/main/java/komus24/model/MinMaxRangeLong.java
package komus24.model;

import komus24.util.StreamUtil;

/**
 * TODO - Document
 */
public class MinMaxRangeLong {
    /**
     * TODO - Document
     */
    private long min;

    /**
     * TODO - Document
     */
    public long getMin() {
        return min;
    }

    /**
     * TODO - Document
     */
    public void setMin(long value) {
        this.min = value;
    }
    /**
     * TODO - Document
     */
    private long max;

    /**
     * TODO - Document
     */
    public long getMax() {
        return max;
    }

    /**
     * TODO - Document
     */
    public void setMax(long value) {
        this.max = value;
    }

    public MinMaxRangeLong(long min, long max) {
        this.min = min;
        this.max = max;
    }

    /**
     * Read MinMaxRangeLong from input stream
     */
    public static MinMaxRangeLong readFrom(java.io.InputStream stream) throws java.io.IOException {
        long min;
        min = StreamUtil.readLong(stream);
        long max;
        max = StreamUtil.readLong(stream);
        return new MinMaxRangeLong(min, max);
    }

    /**
     * Write MinMaxRangeLong to output stream
     */
    public void writeTo(java.io.OutputStream stream) throws java.io.IOException {
        StreamUtil.writeLong(stream, min);
        StreamUtil.writeLong(stream, max);
    }

    /**
     * Get string representation of MinMaxRangeLong
     */
    @Override
    public String toString() {
        StringBuilder stringBuilder = new StringBuilder("MinMaxRangeLong { ");
        stringBuilder.append("min: ");
        stringBuilder.append(String.valueOf(min));
        stringBuilder.append(", ");
        stringBuilder.append("max: ");
        stringBuilder.append(String.valueOf(max));
        stringBuilder.append(" }");
        return stringBuilder.toString();
    }
}

File: clients/swift/Sources/Komus24/Model/MinMaxRangeLong.swift
/// TODO - Document
public struct MinMaxRangeLong {
    /// TODO - Document
    let min: Int64

    /// TODO - Document
    let max: Int64

    /// Read MinMaxRangeLong from input stream
    static func readFrom<S: InputStream>(_ stream: S) -> MinMaxRangeLong {
        var min: Int64
        min = stream.readInt64()
        var max: Int64
        max = stream.readInt64()
        return MinMaxRangeLong(min: min, max: max)
    }

    /// Write MinMaxRangeLong to output stream
    func writeTo<S: OutputStream>(_ stream: S) {
        stream.writeInt64(min)
        stream.writeInt64(max)
    }
}